src/try_filter.rs

```diff
@@ -27,6 +27,27 @@ pub trait TryFilter<O, E>: Sized {
     fn try_filter_ok<F>(self, _: F) -> TryFilterOk<Self, F>
     where
         F: FnMut(&O) -> Result<bool, E>;
+
+    /// Filters every `Err`-value with a function that can return an Err.
+    /// Useful when the filter condition uses functions that can fail.
+    ///
+    ///```
+    /// use resiter::try_filter::TryFilter;
+    /// use std::num::ParseIntError;
+    /// use std::str::FromStr;
+    ///
+    /// let v = ["1", "2", "4", "a", "5"]
+    ///     .iter()
+    ///     .map(|txt| usize::from_str(txt))
+    ///     .try_filter_err(|_:&ParseIntError| Ok(false))
+    ///     .collect::<Vec<Result<_, _>>>();
+    ///
+    /// assert_eq!(v.iter().filter(|x| x.is_ok()).count(), 4);
+    /// assert_eq!(v.iter().filter(|x| x.is_err()).count(), 0);
+    ///```
```

Comment:
```suggestion
    /// ```
    /// use resiter::try_filter::TryFilter;
    /// use std::num::ParseIntError;
    /// use std::str::FromStr;
    ///
    /// let v = ["1", "2", "4", "a", "5"]
    ///     .iter()
    ///     .map(|txt| usize::from_str(txt))
    ///     .try_filter_err(|_:&ParseIntError| Ok(false))
    ///     .collect::<Vec<Result<_, _>>>();
    ///
    /// assert_eq!(v.iter().filter(|x| x.is_ok()).count(), 4);
    /// assert_eq!(v.iter().filter(|x| x.is_err()).count(), 0);
    /// ```
```